hibike/devices/YogiBear/YogiBear.cpp

```diff
@@ -1,122 +1,324 @@
 #include "YogiBear.h"
+#include "ui.h"
+#include "pid.h"
+#include "encoder.h"
+#include "current_limit.h"
+#include "motor.h"
+#include "LED.h"
 
-// each device is responsible for keeping track of it's own params
-uint32_t params[NUM_PARAMS];
+//A space for constants.
+float pwmInput = 0; //Value that is received from hibike and is the goal PWM
+uint8_t driveMode = 0; //0 for manual, 1 for PID Velocity, 2 for PID Position
+bool enabled = false;
 
-/*
-  YOGI BEAR PARAMS:
-    duty = percent duty cycle
-    fault : {1 = normal operation, 0 = fault}
-    forward : {1 = forward, 0 = reverse}
-    inputA : {1 = High, 0 = Low}
-    inputB : {1 = High, 0 = Low}
-*/
+double pos = 0; //these both used to be volatile, but PID library doesn't convert well between volatiles and not-volatiles in c++
+double vel = 0;
 
-uint32_t duty;
-uint32_t fault;
-uint32_t forward;
-uint32_t inputA;
-uint32_t inputB;
+bool hibike = true;
+bool continualPrint = false;
+unsigned long heartbeat = 0;
+unsigned long hibikeHeartbeatLimit = 500;
```

Comment:
This variable doesn't look like it's used anywhere. Please remove or add TODO. (If the functionality to support it is non-trivial, it would be better to do that in a future PR instead of folding it into this one.)